### ISSUE TITLE
NumberBox UI not consistent on Xbox

### ISSUE THREAD

**Issue by under3415** [microsoft/microsoft-ui-xaml#4638]
**Describe the bug**
When using keyboard to tab through controls in Windows 10, user can press up and down arrow keys to increment and decrement numbers in a number box. This does not work on Xbox One. 
If you use D-Pad or sticks to navigate on the page, when you get to a number box you cannot use up and down either on D-Pad or sticks to adjust the value. All you can do is press "GamePad A" which opens a number keyboard. keyboard is very large and complicated to use, so it is not a very good UX. 

**Steps to reproduce the bug**

Use NumberBox on Xbox

**Expected behavior**
When user navigates to a NumberBox she should be able to use up and down controls to quickly adjust the number without opening up the number keyboard

**Screenshots**

**Version Info**
Microsoft.UI.Xaml v2.6.9-prerelease.210227001

NuGet package version: 
Microsoft.UI.Xaml v2.6.9-prerelease.210227001

Windows app type:
| UWP              | Win32            |
| :--------------- | :--------------- |
|  |  |

| Xbox One

| Device form factor | Saw the problem? |
| :----------------- | :--------------- |

| Xbox               |  |

**Additional context**

**Comment by StephenLPeters** [comment]
I think this is probably by design, we don't want to "trap" gamepad users within a control so I don't think we want to change either the joystick or dpad behavior.  It would be nice if there was a way to execute a small or large increment using gamepad... maybe something like the trigger and bumper buttons. Something like right bumper for large increment up and right trigger for small increment up.

**Comment by StephenLPeters** [comment]
@SavoySchuler as the numberbox PM

**Comment by SavoySchuler** [comment]
@chigy Regarding Xbox UX, are there any gamepad input options that would make sense for incrementing/decrementing of NumberBox? As far as I am aware, bumpers, triggers, d-pad, and joystick are all reserved for navigation. Further, I believe input is not intended to be trapped by a control unless "A" has been pressed (aside from z-band elevated UI like a dialog) - is that correct?

**Comment by mdtauk** [comment]
> @chigy Regarding Xbox UX, are there any gamepad input options that would make sense for incrementing/decrementing of NumberBox? As far as I am aware, bumpers, triggers, d-pad, and joystick are all reserved for navigation. Further, I believe input is not intended to be trapped by a control unless "A" has been pressed (aside from z-band elevated UI like a dialog) - is that correct?

@chigy How is the Right Analog Stick treated differently than the Left Stick?  Left would be used for moving, but does Right do the same?

**Comment by under3415** [comment]
On a PC, up and down arrow keys are also used for navigation, but NumberBox uses them for adjusting the numbers. User can escape using tab, or left and right. It's very easy and intuitive.
I don't see why not do the same on Xbox

Also, have a look at FlipView control. It allows just that on Xbox. When you enter control you can flip up and down with sticks, exit using left/right (or vice versa for horizontal one). 

If you cannot use up and down, then you can have a text box, no point in using NumberBox.

**Comment by chigy** [comment]
This is currently by design. See the following guidance documentation.
https://docs.microsoft.com/en-us/windows/uwp/design/input/gamepad-and-remote-interactions#focus-engagement

We have not explored further optimization for Xbox (such as taking advantage of right stick) due to the priority our team has on the matter.

**Comment by under3415** [comment]
Let's forget about trapping users for moment and look at it from a different angle. 

NumberBox has up and down buttons. These buttons are unreachable on Xbox, User cannot use them, ever. Even after pressing "gamepad A" button to activate the control.

Surely you agree this is a problem? Why have the buttons if they are permanently inaccessible! You should either hide them on Xbox or provide a way to use them.

**Comment by chigy** [comment]
@under3415 , I totally read the original post wrong... My apologies...

Yes, it appear to be a problem here. However let me be very open and honest here. As I mentioned, optimizing for Xbox is not our current priority, so our goal is to provide usable solution which this experience does deliver.

For us to do a good job here, we need to revisit and really figure out what's the system approach to using DPad, Joystick, etc. and apply them systematically and consistently across *all* of our controls. This original effort took us months of investigation working with Xbox team as well as figuring out what we need to change on keyboard behavior as well to ensure we can be truly *universal* in our approach. We do not currently have the resources to do a good job on and we would like to avoid introducing on-off solution. 

Because of this, I would suggest that we hide the UP/DOWN flyout on Xbox to avoid user confusion. 
I realize this is not ideal but I think that would be in alignment with what we are doing with rest of the controls.

@SavoySchuler for your input.

**Comment by under3415** [comment]
I feel you are overthinking this. Just copy what FlipView and ListBox are doing. UP/DOWN adjusts the number. LEFT/RIGHT escapes the control. If you really want to be on the safe side, only do this for the joystick and leave D-Pad as navigation only. 

If you remove up/down, you may as well remove NumberBox completely, it becomes a TextBox...
D

**Comment by mdtauk** [comment]
> I feel you are overthinking this. Just copy what FlipView and ListBox are doing. UP/DOWN adjusts the number. LEFT/RIGHT escapes the control. If you really want to be on the safe side, only do this for the joystick and leave D-Pad as navigation only.
> 
> If you remove up/down, you may as well remove NumberBox completely, it becomes a TextBox...
> D

A TextBox which also evaluates math expressions.

But I think ASAP the spin buttons or the increment and decrement functions should be added with gamepad controls.  If the WinUI team are busy, maybe someone from Xbox could offer a suggestion how the interaction should be done.

Once you know how it should work, the community may be able to implement it, or it can be added to a 3.X timeframe.

**Comment by chigy** [comment]
> I feel you are overthinking this. Just copy what FlipView and ListBox are doing. UP/DOWN adjusts the number. LEFT/RIGHT escapes the control. If you really want to be on the safe side, only do this for the joystick and leave D-Pad as navigation only.
> 
> If you remove up/down, you may as well remove NumberBox completely, it becomes a TextBox...
> D

@under3415 , 
It is not that simple. "UP/DOWN adjusts the number. LEFT/RIGHT escapes the control." is only done with the control that we are comfortable user is not going to get stuck. In NumberBox's case, there is a big chance user will get stuck. ListBox working this way is most likely a co-incidence. We have not done any work on this control in many years.

**Comment by chigy** [comment]
> > I feel you are overthinking this. Just copy what FlipView and ListBox are doing. UP/DOWN adjusts the number. LEFT/RIGHT escapes the control. If you really want to be on the safe side, only do this for the joystick and leave D-Pad as navigation only.
> > If you remove up/down, you may as well remove NumberBox completely, it becomes a TextBox...
> > D
> 
> A TextBox which also evaluates math expressions.
> 
> But I think ASAP the spin buttons or the increment and decrement functions should be added with gamepad controls. If the WinUI team are busy, maybe someone from Xbox could offer a suggestion how the interaction should be done.
> 
> Once you know how it should work, the community may be able to implement it, or it can be added to a 3.X timeframe.

@mdtauk ,
As mentioned, doing the type of investigation you are requesting is not a priority for WinUI team nor Xbox team at this time unfortunately.
